src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -3839,6 +3839,7 @@ void MacroAssembler::cmpxchg_narrow_value_helper(Register addr, Register expecte
                                                  enum operand_size size,
                                                  Register shift, Register mask, Register aligned_addr) {
   assert(size == int8 || size == int16, "unsupported operand size");
+  assert(!(UseZacas && UseZabha), "Use amocas");
```

Comment:
Seems more obvious to move this two assertions to its callers?
`MacroAssembler::cmpxchg_narrow_value` and `MacroAssembler::weak_cmpxchg_narrow_value`.